### ISSUE TITLE
from_date greater than 3 years ago gives initially displayed transactions and does not create an error.

### ISSUE THREAD

**Issue by inktrap** [hoffie/dkb-visa#7]
I tested this via dkbweb.py which calls dkb.py, so this should apply to dkb.py as well.

If you specify a from_date like 1.1.1970, you won't get an error from mechanize. But it triggers another issue: The csv export will only contain the initially visible transactions because the date is not submittable. This means that the page won't update and the csv-file won't be generated with the new dates.

I "fixed" this for myself by fetching all the transactions from the last year every time and later I filter the results via sets. See: <https://github.com/inktrap/dkb-export/blob/master/dkbweb.py#L215>. Btw.: I also checked that BASEURL is an https-url. :)

**Comment by hoffie** [comment]
Hm, I am not sure I understand. A quick test revealed that changing the from_date does work for me. I guess there may be DKB-side limits how long you can go back in time...?
So, either I misunderstand or there must be some other difference between your test and mine. :)

Regarding the HTTPS-enforcement: As the URL is hardcoded and not user-changable, I think it is OK to leave it as-is.

**Comment by hoffie** [comment]
Ok, now I see -- the web interface only returns about one year of data, if I see it correctly. Yes, this is a limitation, but I do not see any way to overcome this besides integrating locally "archived" data.

**Comment by okurz** [comment]
well, at least the error handling could be improved on the side of dkb-visa. If I set 01.01.1970 in the current version and submit, the UI automatically sets to the first date possible, 01.06.2015, and tells with a popup field: "Die Eingabe wurde systemseitig angepasst."

**Comment by inktrap** [comment]
Yes, that is the behaviour I ment: dkb falls back to the date where you
they have data. So the user of the script should be informed that they
don't get what they asked for.

Just FYI: In my fork I set the based on todays date minus two years and 11 months.
that way I get all the transactions. Then I wrote some code to
deduplicate the transactions.

Excerpts from Oliver Kurz's message of 2016-05-29 23:01:30 -0700:
